src/xunit.performance.execution/BenchmarkEventSource.cs

```diff
@@ -121,23 +115,27 @@ public unsafe void BenchmarkStop(string RunId, string BenchmarkName, string Stop
                 fixed (char* pBenchmarkName = BenchmarkName)
                 fixed (char* pStopReason = StopReason)
                 {
-                    EventData* data = stackalloc EventData[3];
+                    const int eventDataCount = 3;
+                    EventData* data = stackalloc EventData[eventDataCount];
                     data[0].Size = (RunId.Length + 1) * sizeof(char);
                     data[0].DataPointer = (IntPtr)pRunId;
-                    data[1].Size = (BenchmarkName.Length + 1) * 2;
+                    data[1].Size = (BenchmarkName.Length + 1) * sizeof(char);
                     data[1].DataPointer = (IntPtr)pBenchmarkName;
                     data[2].Size = (StopReason.Length + 1) * 2;
                     data[2].DataPointer = (IntPtr)pStopReason;
-                    WriteEventCore(2, 3, data);
+                    WriteEventCore(2, eventDataCount, data);
                 }
             }
 
             if (_csvWriter != null)
                 WriteCSV(BenchmarkName, stopReason: StopReason);
         }
 
-        [Event(3, Opcode = EventOpcode.Info, Task = Tasks.BenchmarkIterationStart)]
-        public unsafe void BenchmarkIterationStart(string RunId, string BenchmarkName, int Iteration)
+        [Event(3,
+            Level = EventLevel.LogAlways,
+            Opcode = EventOpcode.Start,
+            Task = Tasks.BenchmarkIteration)]
+        public unsafe void BenchmarkIterationStart(string RunId, string BenchmarkName, int Iteration, long AllocatedBytes)
         {
             if (_csvWriter != null)
                 WriteCSV(BenchmarkName, iteration: Iteration);
```

Comment:
I think you probably should.  I suspect that the first thing that will happen after you add this functionality for Windows is that it will be tried on Linux and won't work.  It would be good - unless there is some architectural issue - to complete these types of features for all platforms at once.